src/generic/grid.cpp

```diff
@@ -11337,4 +11400,640 @@ wxGetContentRect(wxSize contentSize,
     return contentRect;
 }
 
+
+#if wxUSE_ACCESSIBILITY
+
+//-----------------------------------------------------------------------------
+// helpers for wxGridAccessible and wxGridCellAccessible
+//-----------------------------------------------------------------------------
+
+// helper to convert row / col into child object id
+// error handling is not required here
+static int GetChildId(wxGrid* grid, int row, int col,
+                      bool isRowHeader, bool isColHeader)
+{
+    int childId = 0;
+    int numberCols = grid->GetNumberCols();
+    if ( grid->GetRowLabelSize() )
+        numberCols++;
+
+    if ( isColHeader && isRowHeader )
+        childId = 1;
+    else if ( isColHeader )
+    {
+        childId = col + 1;
+        if ( grid->GetRowLabelSize() )
+            childId++;
+    }
+    else if ( isRowHeader )
+    {
+        childId = row * numberCols + 1;
+        if ( grid->GetColLabelSize() )
+            childId += numberCols;
+    }
+    else
+    {
+        childId = row * numberCols + col + 1;
+        if ( grid->GetColLabelSize() )
+            childId += numberCols;
+        if ( grid->GetRowLabelSize() )
+            childId++;
+    }
+    return childId;
+}
+
+// helper to calculate row, col, isRowHeader, isColHeader from childId
+static void GetRowCol(wxGrid* grid, int childId, int* row, int* col,
+                      bool* isRowHeader, bool* isColHeader)
+{
+    if ( childId == 1 && grid->GetColLabelSize() && grid->GetRowLabelSize() )
+    {
+        *row = *col = -1;
+        *isRowHeader = true;
+        *isColHeader = true;
+    }
+    else
+    {
+        *isRowHeader = *isColHeader = false;
+        int numCols = grid->GetNumberCols();
+        if ( grid->GetRowLabelSize() )
+            numCols++;
+        *row = (childId - 1) / numCols;
+        *col = (childId - 1) % numCols;
+        if ( grid->GetRowLabelSize() )
+        {
+            if ( *col == 0 )
+            {
+                *col = -1;
+                *isRowHeader = true;
+            }
+            else
+                (*col)--;
+        }
+        if ( grid->GetColLabelSize() )
+        {
+            if ( *row == 0 )
+            {
+                *row = -1;
+                *isColHeader = true;
+            }
+            else
+                (*row)--;
+        }
+    }
+}
+
+
+//-----------------------------------------------------------------------------
+// wxGridAccessible
+//-----------------------------------------------------------------------------
+
+wxGridAccessible::wxGridAccessible(wxGrid* win)
+    : wxWindowAccessible(win)
+{
+    m_gridCellAccessible = nullptr;
+}
+
+wxGridAccessible::~wxGridAccessible()
+{
+    if ( m_gridCellAccessible )
+    {
+        delete m_gridCellAccessible;
+        m_gridCellAccessible = nullptr;
+    }
```

Comment:
All this could be replaced by just `delete m_gridCellAccessible`.

Even better would be to replace the raw pointer with `std::unique_ptr<>`, then it wouldn't need to be neither initialized nor deleted, as it would happen automatically.